### PR TITLE
fix(components):[time-picker] add parseFloat to resolve TimePicker offset

### LINKED ISSUE #12835: [Component] [date-picker, time-picker] 在使用px2rem时，TimePicker偏移（Timer-picker offset with px2rem
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.2`
- Element Plus Version: `2.3.4`
- Browser / OS: `Chrome/113.0 MacOS X 10`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`
- `el-time-picker`

### Reproduction Link

[Github Repo](https://github.com/wan-kong/Element-plus-time-offset)

### Steps to reproduce

1.git clone https://github.com/wan-kong/Element-plus-time-offset
2.pnpm install
3.pnpm run dev
4.设置浏览器宽度小于1920px,例如：1512px(MacBook 14寸的尺寸) (adjust screen width below 1920px,such as 1512px
5.使用时间选择器选择23:59:59（select 23:59:59 with time-picker

### What is Expected?

滚动可以正常适应
(scroll can adjust the height with float number）

### What is actually happening?

滚动位置偏移
（scroll is round the value to an integer)

### Additional comments

我已经找到了解决方案，并且写在了仓库的readme中，如果需要的话，我可以提PR
(I've found the solution and written it in the repo readme, so I can mention PR if I need to

### PR DESCRIPTION
**Pull request by zhengsixsix**
Height is calculated using offsetHeight, offsetHeight will be rounded off

fix #12835

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/time-picker/src/time-picker-com/basic-time-spinner.vue [hunk 1]
```diff
@@ -212,7 +212,14 @@ const adjustSpinner = (type: TimeUnit, value: number) => {
 
 const typeItemHeight = (type: TimeUnit): number => {
   const scrollbar = unref(listRefsMap[type])
-  return scrollbar?.$el.querySelector('li').offsetHeight || 0
+  const listItem = scrollbar?.$el.querySelector('li')
+  if (listItem) {
+    const computedStyle = window.getComputedStyle(listItem)
+    const height = Number.parseFloat(computedStyle.getPropertyValue('height'))
+    return height || 0
```
Comment (btea) [inline packages/components/time-picker/src/time-picker-com/basic-time-spinner.vue:219]:
```suggestion
    return Number.parseFloat(getStyle(listItem, 'height')) || 0
```

https://github.com/element-plus/element-plus/blob/798d9d873939815e750786280c4993d25b05bf59/packages/utils/dom/style.ts#L29-L32

Comment (zhengsixsix) [reply packages/components/time-picker/src/time-picker-com/basic-time-spinner.vue]: have been updated

File: packages/components/time-picker/src/time-picker-com/basic-time-spinner.vue [hunk 2]
```diff
@@ -212,7 +213,12 @@ const adjustSpinner = (type: TimeUnit, value: number) => {
 
 const typeItemHeight = (type: TimeUnit): number => {
   const scrollbar = unref(listRefsMap[type])
-  return scrollbar?.$el.querySelector('li').offsetHeight || 0
+  const listItem = scrollbar?.$el.querySelector('li')
+  if (listItem) {
+    return Number.parseFloat(getStyle(listItem, 'height')) || 0
+  } else {
```
[the remark this answers is not in the document]
Comment (zhengsixsix) [reply packages/components/time-picker/src/time-picker-com/basic-time-spinner.vue]: updated

[the remark this answers is not in the document]
Comment (zhengsixsix) [reply packages/components/time-picker/src/time-picker-com/basic-time-spinner.vue]: I'm sorry, I didn't understand it, but it should meet your needs now.

**Review by btea — APPROVED** [review]
LGTM. Thanks!

**Review by tolking — APPROVED** [review]

**Pull request merged**